### ISSUE TITLE
[stacks-coordinator] improved error handling

### ISSUE THREAD

**Issue by jferrant** [Trust-Machines/core-eng#257]
Right now stacks coordinator pretty much terminates at even the slightest of errors. I do not think we should be terminating by default. Run through the code base to determine when an error should be handled with a warning and parsing continue vs actually truly terminating the coordinator (configs for example should terminate the program. however, failed rpc calls, no.)

- [ ] https://github.com/Trust-Machines/core-eng/issues/256
- [ ] CHECK TO SEE ALL OTHER INSTANCES

**Comment by xoloki** [comment]
Agreed.  Bad config and bad keys should kill on startup.  Bad tx gets a warning.

I have a bunch of `assert!` in the sig verify code that needs to be replaced by sane error handling.
